### PR TITLE
Widget - Support dynamic colors

### PR DESCRIPTION
**Pull request by ashiagr**
## Description

This adds support for dynamic colors for the player widget. 

## Testing Instructions
1. Install the app 
2. Long press the app icon 
3. Tap Widgets
4. Drag the widget to the home screen
5. ✅ Notice that the colors of the button matches those of the wallpaper
6. Change the wallpaper
7. ✅ Notice that the colors of the widget are updated

** Test on Android 11, Android 12, and Android 13 devices (light + dark theme)
** Dynamic colors are not supported below Android 12 so original colors are retained on versions less than Android 12.

## Screenshots or Screencast 

https://github.com/Automattic/pocket-casts-android/assets/1405144/6c59d311-3f44-45f1-90e6-a3395e08d652

Android 9 (API 28) | Android 12 | Android 14 (light theme)
---|---|---
 |  | 

## Checklist
- [ ] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [ ] I have considered whether it makes sense to add tests for my changes
- [ ] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [ ] Any jetpack compose components I added or changed are covered by compose previews
 
#### I have tested any UI changes...

- [x] with different themes
- [ ] with a landscape orientation
- [ ] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

### CODE REVIEW

File: modules/services/images/src/main/res/color/color_primary_container_80.xml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:alpha="0.8" android:color="?attr/colorPrimaryContainer" />
```
Comment (ashiagr) [inline modules/services/images/src/main/res/color/color_primary_container_80.xml:3]: I have kept the background opacity at 80%. 

/cc @david-gonzalez-a8c

**Comment by CookieyedCodes** [comment]
The play icon in my opinion still looks quite dated 😅-tho maby it's because it's big-, any other widgits planned for android?(maby an upnext one?), someone asked me recently could they add a next episode button & considering bookmarks they probably should be added to the widgit 🤔🤔

**Comment by ashiagr** [comment]
This was a good-to-have, straightforward change but there will be more updates coming to widgets. @david-gonzalez-a8c is already on it.

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -263,7 +263,7 @@ jsonassert = "org.skyscreamer:jsonassert:1.5.0"
 junit = "junit:junit:4.13.2"
 junit-ext = "androidx.test.ext:junit:1.1.3"
 leakcanary = "com.squareup.leakcanary:leakcanary-android:2.10"
-material = "com.google.android.material:material:1.5.0"
+material = "com.google.android.material:material:1.9.0"
```
Comment (ashiagr) [inline gradle/libs.versions.toml:266]: I updated material lib to a stable version 1.9.0  for targetSdkVerrsion 33. This was primarily to [prevent private resource warning for colorPrimaryContainer](https://rb.gy/m8okwy).

If you think it is too big bump for this PR, I can update it to 1.6.0 which includes a fix.

**Comment by CookieyedCodes** [comment]
> This was a good-to-have, straightforward change but there will be more updates coming to widgets. @david-gonzalez-a8c is already on it. 

Coolio looking forward to seeing it😀

File: gradle/libs.versions.toml (hunk 2 above)
Comment (mchowning) [reply gradle/libs.versions.toml:266]: Thanks for flagging this. I think it's fine. 👍

**Pull request merged**